source/common/http/filter/transformation_filter.cc

```diff
@@ -153,32 +165,49 @@ void FunctionalTransformationFilter::checkRequestActive() {
 
 void TransformationFilterBase::checkResponseActive() {
   response_transformation_ = getTransformFromRoute(
-      encoder_callbacks_->route(),
-      Config::MetadataTransformationKeys::get().RESPONSE_TRANSFORMATION);
+      encoder_callbacks_->route(), TransformationFilterBase::Direction::Response);
 }
 
 const envoy::api::v2::filter::http::Transformation *
 TransformationFilterBase::getTransformFromRoute(
-    const Router::RouteConstSharedPtr &route, const std::string &key) {
+    const Router::RouteConstSharedPtr &route, TransformationFilterBase::Direction direction) {
 
   if (!route) {
     return nullptr;
   }
 
+  const auto *config = SoloFilterUtility::resolvePerFilterConfig<
+      RouteTransformationFilterConfig>(
+      Config::TransformationFilterNames::get().TRANSFORMATION, route);
+
+  if (config != nullptr) {
+    switch (direction) {
+    case TransformationFilterBase::Direction::Request:
+      return &config->getRequestTranformation();
+    case TransformationFilterBase::Direction::Response:
+      return &config->getResponseTranformation();
+    }
```

Comment:
Should there be a `default` clause?

Reply:
there shouldn't - in general i prefer to avoid defaults in enum switch so the compiler can complain on missing cases.

Reply:
changed to default: NOT_REACHED as it seems the common thing in envoy